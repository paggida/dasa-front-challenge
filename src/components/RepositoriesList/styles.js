import styled from "styled-components";
import metrics from "../../styles/metrics";
import color from "../../styles/colors";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: ${metrics.baseMargin * 5 + metrics.base} 0px
    ${metrics.baseMargin * 4 + metrics.base} 0px;
  min-width: 500px;
  background: ${color.white};
  border-radius: ${metrics.baseRadius + metrics.base};

  header {
    padding: ${metrics.basePadding + 10 + metrics.base};
    display: flex;
    flex-direction: column;
    align-items: center;

    img {
      width: 80px;
      border-radius: 100px;
      padding: 0;
    }

    small {
      font-size: ${metrics.fontSize + 4 + metrics.base};
      color: ${color.dark};
    }
  }
`;

export const Repositories = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;

  ul {
    list-style: none;

    li {
      font-weight: bold;
      padding: ${metrics.basePadding + metrics.base}
        ${metrics.basePadding / 2 + metrics.base};

      strong {
        font-size: ${metrics.fontSize * 2 + metrics.base};
        display: block;
        margin: 0px 0px ${metrics.baseMargin * 2 + metrics.base} 0px;
      }

      small {
        font-weight: normal;
        font-size: ${metrics.fontSize + 2 + metrics.base};
        color: ${color.regular};
        font-style: italic;
      }

      &:nth-child(2n-1) {
        background: ${color.light};
      }
    }
  }
`;
