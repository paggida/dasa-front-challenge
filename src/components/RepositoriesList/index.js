import React from "react";
import PropTypes from "prop-types";
import { Container, Repositories } from "./styles";

const RepositoriesList = ({ repositories }) => {
  if (repositories.length) {
    console.tron.log(repositories[0].owner.avatar_url);
    return (
      <Container>
        <header>
          <img
            src={repositories[0].owner.avatar_url}
            alt={repositories[0].owner.login}
          />
          <small>{repositories[0].owner.login}</small>
        </header>
        <Repositories>
          <ul>
            {repositories.map(repository => (
              <li>
                <strong>{repository.name}</strong>
                {repository.stargazers_count} <small>stars</small>
              </li>
            ))}
          </ul>
        </Repositories>
      </Container>
    );
  } else {
    return <div />;
  }
};

RepositoriesList.propTypes = {
  repositories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      owner: PropTypes.shape({
        avatar_url: PropTypes.string,
        login: PropTypes.string
      }),
      name: PropTypes.string,
      stargazers_count: PropTypes.number
    })
  ).isRequired
};

export default RepositoriesList;
