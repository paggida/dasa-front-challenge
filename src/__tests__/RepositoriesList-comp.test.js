import React from 'react';
import RepositoriesList from '../components/RepositoriesList';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json"


it("Should render the repositories list for an empty repository group", () => {
  const props = {
    repositories: []
  };
  const comp = mount(<RepositoriesList {...props} />);
  expect(toJson(comp)).toMatchSnapshot();
  comp.unmount();
});

it("Should render the list of repositories for a populated repository group", () => {
  const props = {
    repositories: [
      { id: 1,
        name: "Repo01",
        stargazers_count: 11,
        owner: {
          login:"Tester01",
          avatar_url:"https://avatars2.githubusercontent.com/u/69631?s=200&v=4"
        }
      },
      { id: 2,
        name: "Repo02",
        stargazers_count: 12,
        owner: {
          login:"Tester02",
          avatar_url:"https://avatars2.githubusercontent.com/u/69631?s=200&v=4"
        }
      },
    ]
  };
  const comp = mount(<RepositoriesList {...props} />);
  expect(toJson(comp)).toMatchSnapshot();
  comp.unmount();
});
