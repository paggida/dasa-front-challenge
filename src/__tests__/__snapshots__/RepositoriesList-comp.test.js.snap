// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the list of repositories for a populated repository group 1`] = `
<RepositoriesList
  repositories={
    Array [
      Object {
        "id": 1,
        "name": "Repo01",
        "owner": Object {
          "avatar_url": "https://avatars2.githubusercontent.com/u/69631?s=200&v=4",
          "login": "Tester01",
        },
        "stargazers_count": 11,
      },
      Object {
        "id": 2,
        "name": "Repo02",
        "owner": Object {
          "avatar_url": "https://avatars2.githubusercontent.com/u/69631?s=200&v=4",
          "login": "Tester02",
        },
        "stargazers_count": 12,
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-AxjAm ggwBWc"
    >
      <header>
        <img
          alt="Tester01"
          src="https://avatars2.githubusercontent.com/u/69631?s=200&v=4"
        />
        <small>
          Tester01
        </small>
      </header>
      <styled.div>
        <div
          className="sc-AxirZ ccREzF"
        >
          <ul>
            <li
              key="1"
            >
              <strong>
                Repo01
              </strong>
              11
               
              <small>
                stars
              </small>
            </li>
            <li
              key="2"
            >
              <strong>
                Repo02
              </strong>
              12
               
              <small>
                stars
              </small>
            </li>
          </ul>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RepositoriesList>
`;

exports[`Should render the repositories list for an empty repository group 1`] = `
<RepositoriesList
  repositories={Array []}
>
  <div />
</RepositoriesList>
`;
